pkg/kubectl/cmd/create_deamonset.go

```diff
@@ -0,0 +1,107 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/kubectl"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/util/i18n"
+)
+
+var (
+	daemonsetLong = templates.LongDesc(i18n.T(`
+	Create a daemonset with the specified name.`))
+
+	daemonsetExample = templates.Examples(i18n.T(`
+	# Create a new daemonset named my-dea that runs the busybox image.
+	kubectl create daemonset my-dea --image=busybox`))
+)
+
+// NewCmdCreateDaemonset is a command to create a new daemonset.
+func NewCmdCreateDaemonset(f cmdutil.Factory, cmdOut, cmdErr io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "daemonset NAME --image=image [--dry-run]",
+		Aliases: []string{"ds"},
+		Short:   i18n.T("Create a daemonset with the specified name."),
+		Long:    daemonsetLong,
+		Example: daemonsetExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(CreateDaemonset(f, cmdOut, cmdErr, cmd, args))
+		},
+	}
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddPrinterFlags(cmd)
+	cmdutil.AddGeneratorFlags(cmd, cmdutil.DaemonsetAppsV1Beta2GeneratorName)
+	cmd.Flags().StringSlice("image", []string{}, "Image name to run.")
+	cmd.MarkFlagRequired("image")
+	return cmd
+}
+
+func CreateDaemonset(f cmdutil.Factory, cmdOut, cmdErr io.Writer, cmd *cobra.Command, args []string) error {
+	name, err := NameFromCommandArgs(cmd, args)
+	if err != nil {
+		return err
+	}
+	generatorName := cmdutil.GetFlagString(cmd, "generator")
+
+	clientset, err := f.ClientSet()
+	if err != nil {
+		return err
+	}
+	resourcesList, err := clientset.Discovery().ServerResources()
+	// ServerResources ignores errors for old servers do not expose discovery
+	if err != nil {
+		return fmt.Errorf("failed to discover supported resources: %v", err)
+	}
+
+	// It is possible we have to modify the user-provided generator name if
+	// the server does not have support for the requested generator.
+	generatorName = cmdutil.FallbackGeneratorNameIfNecessary(generatorName, resourcesList, cmdErr)
+
+	var generator kubectl.StructuredGenerator
+	switch generatorName {
+	case cmdutil.DaemonsetExtensionsV1Beta1GeneratorName:
+		generator = &kubectl.DaemonSetGeneratorExtensionsV1Beta1{
+			BaseGenerator: kubectl.BaseGenerator{
+				Name:   name,
+				Images: cmdutil.GetFlagStringSlice(cmd, "image"),
```

Comment:
Both L88 and L95 use this variable. Move `cmdutil.GetFlagStringSlice(cmd, "image")` outer?